crates/augurs-forecaster/src/transforms.rs

```diff
@@ -371,6 +392,90 @@ trait ExpExt: Iterator<Item = f64> {
 
 impl<T> ExpExt for T where T: Iterator<Item = f64> {}
 
+/// Returns the Box-Cox transformation of the given value.
+/// Assumes x > 0.
+pub fn box_cox(x: f64, lambda: f64) -> f64 {
+    if x <= 0.0 {
+        panic!("Input x must be positive for Box-Cox transformation.");
+    }
+    if lambda == 0.0 {
+        x.ln()
+    } else {
+        (x.powf(lambda) - 1.0) / lambda
+    }
+}
+
+/// An iterator adapter that applies the Box-Cox transformation to each item.
+#[derive(Clone, Debug)]
+struct BoxCox<T> {
+    inner: T,
+    lambda: f64,
+}
+
+impl<T> Iterator for BoxCox<T>
+where
+    T: Iterator<Item = f64>,
+{
+    type Item = f64;
+    fn next(&mut self) -> Option<Self::Item> {
+        self.inner.next().map(|x| box_cox(x, self.lambda))
+    }
+}
+
+trait BoxCoxExt: Iterator<Item = f64> {
+    fn boxcox(self, lambda: f64) -> BoxCox<Self>
+    where
+        Self: Sized,
+    {
+        BoxCox {
+            inner: self,
+            lambda,
+        }
+    }
+}
+
+impl<T> BoxCoxExt for T where T: Iterator<Item = f64> {}
+
+/// Returns the inverse Box-Cox transformation of the given value.
+fn inverse_box_cox(y: f64, lambda: f64) -> f64 {
+    if lambda == 0.0 {
+        y.exp()
+    } else {
+        (y * lambda + 1.0).powf(1.0 / lambda)
+    }
+}
+
+/// An iterator adapter that applies the inverse Box-Cox transformation to each item.
+#[derive(Clone, Debug)]
+struct InverseBoxCox<T> {
+    inner: T,
+    lambda: f64,
+}
+
+impl<T> Iterator for InverseBoxCox<T>
+where
+    T: Iterator<Item = f64>,
+{
+    type Item = f64;
+    fn next(&mut self) -> Option<Self::Item> {
+        self.inner.next().map(|y| inverse_box_cox(y, self.lambda))
+    }
+}
+
+trait InverseBoxCoxExt: Iterator<Item = f64> {
+    fn inverse_boxcox(self, lambda: f64) -> InverseBoxCox<Self>
```

Comment:
```suggestion
    fn inverse_boxcox(self, lambda: f64) -> InverseBoxCox<Self>
```

same re. consistency